### PR TITLE
MINOR: Combine repeated top-level error test in AlterIsrManagerTest

### PR DESCRIPTION
**Pull request by dengziming**
The content of the 3 top-level errors are identical, just merge them.

### CODE REVIEW

**Comment by dengziming** [comment]
@cmccabe  hi, PTAL.

**Comment by dengziming** [comment]
@mumrah  Hi, this class is created by you so I think you can have a look, thank you.

**Comment by dengziming** [comment]
@mumrah Hi, David, PTAL.

**Review by chia7712 — COMMENTED** [review]
@dengziming Thanks for your patch. It seems to me the name of test case is good to clarify the purpose. Another way to remove duplicate code is that we can add a private helper method to handle duplicate parts. For example:
```scala
def testTopLevelError(error: Error): Unit = {
  val sirs = ...
  val manager = ...
  assertFalse(...
}

@Test
def testOtherErrors(): Unit = {
  testTopLevelError(Errors.UNKNOWN_SERVER_ERROR)
}
``` 

Each test case can call the helper method to complete testing. WDYT?

**Comment by dengziming** [comment]
@chia7712 ,Thank you for your suggestion. In fact, I get the idea from `AlterIsrManagerTest.testPartitionErrors` which test 3 errors in a method, so I also merge the 3 top errors in a method. I think you offer a better idea and I also changed the 3 partition level error.

File: core/src/test/scala/unit/kafka/server/AlterIsrManagerTest.scala [hunk 1]
```diff
@@ -127,26 +127,24 @@ class AlterIsrManagerTest {
...
+    testTopLevelError(Errors.STALE_BROKER_EPOCH)
   }
 
   @Test
-  def testOtherErrors(): Unit = {
+  def testUnknownServer(): Unit = {
+    testTopLevelError(Errors.UNKNOWN_SERVER_ERROR)
+  }
+
+  def testTopLevelError(error: Errors): Unit = {
```
Comment (mumrah) [inline core/src/test/scala/unit/kafka/server/AlterIsrManagerTest.scala]: Maybe rename this and "testPartitionError" to something slightly different so they don't look like actual test cases? Maybe something like "assertTopLevelError" or "checkTopLevelError".

**Review by mumrah — COMMENTED** [review]
@dengziming thanks for the patch. This cleanup looks fine, just one comment inline.

File: core/src/test/scala/unit/kafka/server/AlterIsrManagerTest.scala [hunk 2]
```diff
@@ -127,26 +127,24 @@ class AlterIsrManagerTest {
...
+    checkTopLevelError(Errors.STALE_BROKER_EPOCH)
   }
 
   @Test
-  def testOtherErrors(): Unit = {
+  def testUnknownServer(): Unit = {
+    checkTopLevelError(Errors.UNKNOWN_SERVER_ERROR)
+  }
+
+  def checkTopLevelError(error: Errors): Unit = {
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/server/AlterIsrManagerTest.scala]: Could you change it to ```private``` method?

File: core/src/test/scala/unit/kafka/server/AlterIsrManagerTest.scala [hunk 3]
```diff
@@ -174,13 +172,24 @@ class AlterIsrManagerTest {
...
+  def testUnknownTopicPartition(): Unit = {
+    checkPartitionError(Errors.UNKNOWN_TOPIC_OR_PARTITION)
+  }
+
+  @Test
+  def testNotLeaderOrFollower(): Unit = {
+    checkPartitionError(Errors.NOT_LEADER_OR_FOLLOWER)
+  }
+
+  def checkPartitionError(error: Errors): Unit = {
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/server/AlterIsrManagerTest.scala]: ditto

Comment (dengziming) [reply core/src/test/scala/unit/kafka/server/AlterIsrManagerTest.scala]: Thanks, done!

Comment (dengziming) [reply core/src/test/scala/unit/kafka/server/AlterIsrManagerTest.scala]: Thanks, done.

**Comment by chia7712** [comment]
@dengziming Could you trigger QA again?

**Comment by dengziming** [comment]
@chia7712 I tried to rebase my pr and push it again, but still failed. this is weird, maybe I triggered a bug of jenkins and should tried again tomorrow.

**Review by chia7712 — APPROVED** [review]
@dengziming Thanks for your patch. LGTM

**Comment by chia7712** [comment]
I will merge this PR tomorrow if there is no objection. @mumrah FYI

**Review by mumrah — APPROVED** [review]
LGTM, thanks!

**Pull request merged**